app/src/main/java/com/dmonzonis/nookpendium/Record.kt

```diff
@@ -18,13 +18,39 @@ data class Record(
     var captured: Boolean?
 )
 
-class Recordset(val records: List<Record>) {
-    fun filter(filter: (r: Record) -> Boolean): List<Record> {
-        return records.filter(filter)
+class Recordset(private val totalRecords: List<Record>) {
+    // Current subset of records after having applied filters/sorting
+    var records = totalRecords
+
+    fun applyFilter(filter: (r: Record) -> Boolean): List<Record> {
+        records = records.filter(filter)
+        return records
+    }
+
+    fun applyFilterByMonth(month: Int): List<Record> {
+        return applyFilter { month < it.availability.size && it.availability[month] == 1 }
+    }
+
+    fun applySort(field: String, descending: Boolean): List<Record> {
+        records = when (field) {
+            "name" -> if (descending) {
+                records.sortedByDescending { it.name }
+            } else {
+                records.sortedBy { it.name }
+            }
+            // Sort by price by default
+            else -> if (descending) {
+                records.sortedByDescending { it.price }
+            } else {
+                records.sortedBy { it.price }
+            }
```

Comment:
Eventually we should add sort by availability, this + the filter could allow you to see which ones you should prioritize